viadot/sources/epicor.py

```diff
@@ -1,11 +1,178 @@
+    Handling: Optional[str]
+    Delivery: Optional[str]
+    Pickup: Optional[str]
+    Restocking: Optional[str]
+    MinimumCharge: Optional[str]
+    DiscountAllowance: Optional[str]
+    SalesTax: Optional[str]
+    TotalInvoice: Optional[str]
+
+
+class HeaderInformation(BaseModel):
+    CompanyNumber: Optional[str]
+    OrderNumber: Optional[str]
+    InvoiceNumber: Optional[str]
+    CustomerNumber: Optional[str]
+    CustomerDescription: Optional[str]
+    CustomerPurchaseOrderNumber: Optional[str]
+    Contact: Optional[str]
+    SellingWarehouse: Optional[str]
+    ShippingWarehouse: Optional[str]
+    ShippingMethod: Optional[str]
+    PaymentTerms: Optional[str]
+    PaymentTermsDescription: Optional[str]
+    FreightTerms: Optional[str]
+    FreightTermsDescription: Optional[str]
+    SalesRepOne: Optional[str]
+    SalesRepOneDescription: Optional[str]
+    EntryDate: Optional[str]
+    OrderDate: Optional[str]
+    RequiredDate: Optional[str]
+    ShippedDate: Optional[str]
+    InvoiceDate: Optional[str]
+    ShipToAddress: Optional[ShipToAddress]
+    TrackingNumbers: Optional[TrackingNumbers]
+    InvoiceTotals: Optional[InvoiceTotals]
+
+
+class LineItemDetail(BaseModel):
+    ProductNumber: Optional[str]
+    ProductDescription1: Optional[str]
+    ProductDescription2: Optional[str]
+    CustomerProductNumber: Optional[str]
+    LineItemNumber: Optional[str]
+    QuantityOrdered: Optional[str]
+    QuantityShipped: Optional[str]
+    QuantityBackordered: Optional[str]
+    Price: Optional[str]
+    UnitOfMeasure: Optional[str]
+    ExtendedPrice: Optional[str]
+    QuantityShippedExtension: Optional[str]
+    LineItemShipWarehouse: Optional[str]
+
+
+class Order(BaseModel):
+    HeaderInformation: Optional[HeaderInformation]
+    LineItemDetail: Optional[LineItemDetail]
+
+
+def parse_orders_xml(xml_data: str) -> pd.DataFrame:
+    """
+    Function to parse xml containing Epicor Orders Data.
+
+    Args:
+        xml_data (str, required): Response from Epicor API in form of xml
+    Returns:
+        pd.DataFrame: DataFrame containing parsed orders data.
+    """
+    final_df = pd.DataFrame()
+    ship_dict = {}
+    invoice_dict = {}
+    header_params_dict = {}
+    item_params_dict = {}
+
+    root = ET.fromstring(xml_data.text)
+
+    for order in root.findall("Order"):
+        for header in order.findall("HeaderInformation"):
+            for tracking_numbers in header.findall("TrackingNumbers"):
+                numbers = ""
+                for tracking_number in tracking_numbers.findall("TrackingNumber"):
+                    numbers = numbers + "'" + tracking_number.text + "'"
+                result_numbers = TrackingNumbers(TrackingNumber=numbers)
+
+            for shipto in header.findall("ShipToAddress"):
+                for ship_param in ShipToAddress.__dict__.get("__annotations__"):
+                    try:
+                        ship_value = shipto.find(f"{ship_param}").text
```

Comment:
f"{ship_param}" this is the same as just `ship_param` (or `str(ship_param)` if it needs to be casted)